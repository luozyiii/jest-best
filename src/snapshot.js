export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    time: new Date(),
  };
};

export const generateAnotherConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    time: new Date(),
  };
};
