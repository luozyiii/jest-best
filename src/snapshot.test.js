import { generateConfig, generateAnotherConfig } from './snapshot';

test('测试 generateConfig 函数', () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});

test('测试 generateAnotherConfig 函数', () => {
  expect(generateAnotherConfig()).toMatchSnapshot();
});

test('测试 generateAnotherConfig 函数2', () => {
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "domain": "localhost",
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `,
  );
});
