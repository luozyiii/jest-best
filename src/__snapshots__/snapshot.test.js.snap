// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateAnotherConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": 2021-06-07T07:13:08.089Z,
}
`;

exports[`测试 generateConfig 函数 1`] = `
Object {
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
